homeassistant/components/unifi/controller.py

```diff
@@ -367,6 +349,19 @@ async def async_config_entry_updated(
         controller.load_config_entry_options()
         async_dispatcher_send(hass, controller.signal_options_update)
 
+    @callback
+    def start_websocket(self) -> None:
+        """Start up connection to websocket."""
+
+        async def _websocket_runner() -> None:
+            """Start websocket."""
+            await self.api.start_websocket()
+            self.available = False
```

Comment:
Shouldnt this be wrapped in try, finally clause? When task is cancelled (or other exception occurs) it will not set available=false

Reply:
It's still the same well tested code underneath and there are a lot of exception handling in it already.

Reply:
Yes, but nothing in that will set the variable available=false. When cancelled, it wont be set, nor will signal be sent. Im not sure its a problem.

Reply:
There is only cancelling happening when reloading or shutting down the integration, apart from that the method should just return if there where any exceptions in the websocket handling of the library

Reply:
I would have had a try finally check so stray exceptions still make sure entities goes unavailable. But im okey with this explenation too.